aws/src/main/java/org/apache/iceberg/aws/glue/DynamoLockManager.java

```diff
@@ -0,0 +1,190 @@
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableRequest;
+import software.amazon.awssdk.services.dynamodb.model.DescribeTableResponse;
+import software.amazon.awssdk.services.dynamodb.model.KeySchemaElement;
+import software.amazon.awssdk.services.dynamodb.model.KeyType;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+import software.amazon.awssdk.services.dynamodb.model.ResourceNotFoundException;
+import software.amazon.awssdk.services.dynamodb.model.ScalarAttributeType;
+import software.amazon.awssdk.services.dynamodb.model.TableStatus;
+
+/**
+ * DynamoDB implementation for the lock manager.
+ */
+class DynamoLockManager implements LockManager {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DynamoLockManager.class);
+
+  private static final String LOCK_TABLE_COL_LOCK_ID = "lockId";
+  private static final String LOCK_TABLE_COL_EXPIRE_TS_MILLIS = "expireTimestampMillis";
+  private static final long LOCK_TABLE_CREATION_WAIT_MS = 5000;
+  private static final int LOCK_TABLE_CREATION_WAIT_ROUNDS = 10;
+
+  private static final List<KeySchemaElement> LOCK_TABLE_SCHEMA = Lists.newArrayList(
+      KeySchemaElement.builder()
+          .attributeName(LOCK_TABLE_COL_LOCK_ID)
+          .keyType(KeyType.HASH)
+          .build()
+  );
+
+  private static final List<AttributeDefinition> LOCK_TABLE_COL_DEFINITIONS = Lists.newArrayList(
+      AttributeDefinition.builder()
+          .attributeName(LOCK_TABLE_COL_LOCK_ID)
+          .attributeType(ScalarAttributeType.S)
+          .build()
+  );
+
+  private final DynamoDbClient dynamo;
+  private final AwsProperties awsProperties;
+
+  DynamoLockManager(DynamoDbClient dynamo, AwsProperties awsProperties) {
+    this.dynamo = dynamo;
+    this.awsProperties = awsProperties;
+    ensureLockTableExistsOrCreate();
+  }
+
+  private void ensureLockTableExistsOrCreate() {
+    ensureTableExistsOrCreate(awsProperties.glueCatalogLockTable(), LOCK_TABLE_SCHEMA, LOCK_TABLE_COL_DEFINITIONS);
+  }
+
+  private void ensureTableExistsOrCreate(String tableName, List<KeySchemaElement> schema,
+                                         List<AttributeDefinition> definitions) {
+    try {
+      dynamo.describeTable(DescribeTableRequest.builder()
+          .tableName(tableName)
+          .build());
+    } catch (ResourceNotFoundException e) {
+      LOG.info("Glue lock DynamoDB table <{}> not found, trying to create", tableName);
+      dynamo.createTable(CreateTableRequest.builder()
+          .tableName(tableName)
+          .keySchema(schema)
+          .attributeDefinitions(definitions)
+          .billingMode(BillingMode.PAY_PER_REQUEST)
+          .build());
+
+      boolean isTableActive = false;
+      int round = 0;
+      while (!isTableActive && round < LOCK_TABLE_CREATION_WAIT_ROUNDS) {
+        LOG.info("waiting for DynamoDB table <{}> to be active, round {}", tableName, round++);
```

Comment:
I don't think that "round" is very clear. In most places we refer to this as an "attempt" or "try". What do you think about using "attempt" here?

Also, could this use `Tasks` instead of custom logic? That supports retries with a configurable backoff policy.